### ISSUE TITLE
Dialog: HTML validation error within footer

### ISSUE THREAD

**Issue by miedrl** [primefaces/primefaces#9132]
### Describe the bug

For accessibility the syntax of the HTML code must be correct. When implementing a p:dialog with a footer facet containing div elements, the HTML validation shows an error message.

### Reproducer

1. Open the example from the reproducer in your browser.
2. Copy the HTML code from the example to the [Nu HTML checker](https://validator.w3.org/nu/#textarea)
3. Click the 'check' button.
4. The result of the Nu HTML checker can be further filtered for accessibility issues by using the following JavaScript code as a bookmarklet:
`javascript:(function(){var removeNg=true;var filterStrings=["tag seen","Stray end tag","Bad start tag","violates nesting rules","Duplicate ID","first occurrence of ID","Unclosed element","not allowed as child of element","unclosed elements","not allowed on element","unquoted attribute value","Duplicate attribute"];var filterRE,root,results,result,resultText,i,cnt=0;filterRE=filterStrings.join("|");root=document.getElementById("results");if(!root){alert("No results container found.");return}results=root.getElementsByTagName("li");for(i=0;i<results.length;i++){result=results[i];if(result.className!==""){resultText=(result.innerText!==undefined?result.innerText:result.textContent)+"";if(resultText.match(filterRE)===null){result.style.display="none";result.className=result.className+" steveNoLike";cnt++}else if(removeNg==true){if(resultText.indexOf("not allowed on element")!==-1&&resultText.indexOf("ng-")!==-1){result.style.display="none";result.className=result.className+" steveNoLike";cnt++}}}}alert("Complete. "+cnt+" items removed.")})();`
5. See error like this
![grafik](https://user-images.githubusercontent.com/77431692/186375478-77c004ee-e759-43dd-81e0-f7807cdd874f.png)

### Expected behavior

There should be no errors when validating the HTML code with the Nu HTML Checker.

The error comes from the following code within the dialog footer. The span element there has to be a div element so that further div elements can be put into the footer.
![grafik](https://user-images.githubusercontent.com/77431692/186375854-6ae9bbab-31d5-43f2-ab86-003ef3fdcf53.png)

### PrimeFaces version

12.0.0

### JSF implementation

All

### JSF version

x

### Java version

x

**Comment by melloware** [comment]
When I check it it is fine for me?

I copied the dialog HTML with footer and pasted in NuHtml checker:

```xml
<!DOCTYPE html>
<html lang="">
<head>
<title>Test</title>
</head>
<body>
<div id="j_idt351" class="ui-dialog ui-widget ui-widget-content ui-corner-all ui-shadow ui-hidden-container ui-draggable ui-resizable" style="width: 350px; height: auto; visibility: visible; left: 776.5px; top: 0px; z-index: 1006; display: block;" role="dialog" aria-describedby="j_idt351_content" aria-hidden="false" aria-labelledby="j_idt351_title" aria-live="polite"><div class="ui-dialog-titlebar ui-widget-header ui-helper-clearfix ui-corner-top ui-draggable-handle"><span id="j_idt351_title" class="ui-dialog-title">Header</span><a href="#" class="ui-dialog-titlebar-icon ui-dialog-titlebar-close ui-corner-all" aria-label="Close" role="button"><span class="ui-icon ui-icon-closethick"></span></a></div><div class="ui-dialog-content ui-widget-content" id="j_idt351_content" style="height: auto;">
            <p class="m-0">Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
            Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
            Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
            cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p></div><div class="ui-dialog-footer ui-widget-content"><span><button id="j_idt353" name="j_idt353" class="ui-button ui-widget ui-state-default ui-corner-all ui-button-text-only ui-button-flat mr-2" style="width: 6rem" type="button"><span class="ui-button-text ui-c">No</span></button><button id="j_idt354" name="j_idt354" class="ui-button ui-widget ui-state-default ui-corner-all ui-button-text-only" style="width: 5rem" type="button"><span class="ui-button-text ui-c">Yes</span></button></span></div><div class="ui-resizable-handle ui-resizable-n" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-s" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-e" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-w" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-ne" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-nw" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-se ui-icon ui-icon-gripsmall-diagonal-se" style="z-index: 90;"></div><div class="ui-resizable-handle ui-resizable-sw" style="z-index: 90;"></div></div>
</body>
</html>
```

it reports..
![image](https://user-images.githubusercontent.com/4399574/186410423-2fcd78c8-d283-4cc3-b1dd-b04a179fe6fd.png)

**Comment by miedrl** [comment]
@melloware As I wrote in the description of the bug, the footer has to contain div elements. You cannot use the dialog showcase code for that.

**Comment by melloware** [comment]
Yep fixed it for 12.0.0 final
